src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -288,67 +258,82 @@ class PopoverReportActionContextMenu extends React.Component {
-                    confirmText={this.props.translate('common.delete')}
-                    cancelText={this.props.translate('common.cancel')}
-                    danger
+    const showDeleteModal = (reportID, reportAction, shouldSetModalVisibility = true, onConfirm = () => {}, onCancel = () => {}) => {
+        onCancelDeleteModal.current = onCancel;
+        onComfirmDeleteModal.current = onConfirm;
+
+        reportIDRef.current = reportID;
+        reportActionRef.current = reportAction;
+
+        setShouldSetModalVisibilityForDeleteConfirmation(shouldSetModalVisibility);
+        setIsDeleteCommentConfirmModalVisible(true);
+    };
+
+    useImperativeHandle(ref, () => ({
+        showContextMenu,
+        hideContextMenu,
+        showDeleteModal,
+        hideDeleteModal,
+        isActiveReportAction,
+        instanceID,
+        runAndResetOnPopoverHide,
+        clearActiveReportAction,
+    }));
+
+    const reportAction = reportActionRef.current;
+
+    return (
+        <>
+            <PopoverWithMeasuredContent
+                isVisible={isPopoverVisible}
+                onClose={hideContextMenu}
+                onModalShow={runAndResetOnPopoverShow}
+                onModalHide={runAndResetOnPopoverHide}
+                anchorPosition={popoverAnchorPosition.current}
+                animationIn="fadeIn"
+                disableAnimation={false}
+                animationOutTiming={1}
+                shouldSetModalVisibility={false}
+                fullscreen
+                withoutOverlay
+                anchorRef={anchorRef}
+            >
+                <BaseReportActionContextMenu
+                    isVisible
+                    type={typeRef.current}
+                    reportID={reportIDRef.current}
+                    reportActionID={reportActionIDRef.current}
+                    draftMessage={reportActionDraftMessageRef.current}
+                    selection={selectionRef.current}
+                    isArchivedRoom={isRoomArchived}
+                    isChronosReport={isChronosReportEnabled}
+                    isPinnedChat={isChatPinned}
+                    isUnreadChat={hasUnreadMessages}
+                    anchor={contextMenuTargetNode}
+                    contentRef={contentRef}
+                    originalReportID={originalReportIDRef.current}
                 />
-            </>
-        );
-    }
+            </PopoverWithMeasuredContent>
+            <ConfirmModal
+                title={translate('reportActionContextMenu.deleteAction', {reportAction})}
+                isVisible={isDeleteCommentConfirmModalVisible}
+                shouldSetModalVisibility={shouldSetModalVisibilityForDeleteConfirmation}
+                onConfirm={confirmDeleteAndHideModal}
+                onCancel={hideDeleteModal}
+                onModalHide={() => {
+                    reportIDRef.current = '0';
+                    reportActionRef.current = {};
```

Comment:
I'm not sure why this was added. 
@OlimpiaZurek Can you explain?

Reply:
@s-alves10 This shouldn't be here. I will prepare a PR to fix this.

Reply:
Link to PR: https://github.com/Expensify/App/pull/28462